src/legilogic_ethereum/ethereum_user.ml

```diff
@@ -368,12 +405,49 @@ let send_and_confirm_transaction : (Transaction.t * SignedTransaction.t, Transac
         if ethereum_user_log then
           Logging.log "sender_nonce=%s nonce=%s" (Revision.to_string sender_nonce) (Revision.to_string nonce);
         if Nonce.(compare sender_nonce nonce > 0) then
-          confirmed_or_known_issue sender hash
+          (if ethereum_user_log then
+             Logging.log "Before call of confirmed_or_known_issue";
+           confirmed_or_known_issue sender hash)
         else
-          fail Still_pending)
-
+          (if ethereum_user_log then
+             Logging.log "Before call of fail Still_pending";
+           fail Still_pending))
     >>= check_receipt_sufficiently_confirmed
 
+
+
+
+
+
+
+let infinite_loop_original : (Transaction.t * SignedTransaction.t) -> TransactionReceipt.t Lwt_exn.t =
```

Comment:
What is this function for? It needs a better name than that.

Reply:
Eliminated and put back the code as before.